kops/kops_test.go

```diff
@@ -20,44 +23,88 @@ type testCase struct {
 	found bool
 }
 
-// FIXME - Put this back in when we figure out what to do about runing Cmd.Run() native
-//func TestCreateCluster(t *testing.T)  {
-//	k, _ := NewKops()
-//	k.envs = [][]string {{"key1", "value1"}, {"key2", "value2"}}
-//	values := []testCase{
-//		{"/usr/local/bin/docker", false},
-//		{"run", false},
-//		{ "-e", false},
-//		{ "key1=value1", false},
-//		{ "key2=value2", false},
-//		{"create", false},
-//		{"cluster", false},
-//		{"--name=" + kopsConfig.Name, false},
-//		{"--state=" + kopsConfig.StateStore, false},
-//		{"--ssh-public-key=" + k.publicKey, false},
-//		{"--vpc=" + kopsConfig.Vpc, false},
-//		{"--master-count=" + strconv.Itoa(kopsConfig.MasterCount), false},
-//		{"--master-size=" + kopsConfig.MasterEc2, false},
-//		{"--node-count=" + strconv.Itoa(kopsConfig.WorkerCount), false},
-//		{"--node-size=" + kopsConfig.WorkerEc2, false},
-//		{"--zones=" + strings.Join(kopsConfig.Zones, ","), false},
-//	}
-//
-//	cmd, _ := k.CreateCluster(context.TODO(), kopsConfig)
-//	cmdString := cmd.GetCmdString()
-//
-//	for _, c := range cmdString {
-//		for i, v := range values {
-//			if (v.value == c) {
-//				values[i].found = true
-//				break
-//			}
-//		}
-//	}
-//
-//	for _, v := range values {
-//		if (v.found == false) {
-//			t.Error("Expected ", v.value, "not found")
-//		}
-//	}
-//}
+var cmd string
+
+func mockRunStreamingCmd(cmdString string) error {
+	cmd = cmdString
+	return nil
+}
+
+func mockRunCmd(cmdString string) (*bytes.Buffer, error) {
+	cmd = cmdString
+	return nil, nil
+}
+
+func TestCreateCluster(t *testing.T) {
+	k, err := NewKops()
+	if err != nil {
+		t.Error("Expected no error got", err)
+		return
+	}
+
+	k.runStreamingCmd = mockRunStreamingCmd
+	k.runCmd = mockRunCmd
+
+	//values := []testCase{
+	//	{"/usr/local/bin/docker", false},
+	//	{"run", false},
+	//	{ "-e", false},
+	//	{ "key1=value1", false},
+	//	{ "key2=value2", false},
+	//	{"create", false},
+	//	{"cluster", false},
+	//	{"--name=" + kopsConfig.Name, false},
+	//	{"--state=" + kopsConfig.StateStore, false},
+	//	{"--ssh-public-key=" + k.publicKey, false},
+	//	{"--vpc=" + kopsConfig.Vpc, false},
+	//	{"--master-count=" + strconv.Itoa(kopsConfig.MasterCount), false},
+	//	{"--master-size=" + kopsConfig.MasterEc2, false},
+	//	{"--node-count=" + strconv.Itoa(kopsConfig.WorkerCount), false},
+	//	{"--node-size=" + kopsConfig.WorkerEc2, false},
+	//	{"--zones=" + strings.Join(kopsConfig.Zones, ","), false},
+	//}
+
+	values := []testCase{
+		{"run", false},
```

Comment:
these test cases can also be updated when kops.go from master is merged since the command string will look a bit different